src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -2122,6 +2128,7 @@ class ObservationStore(
       isPermanent: Boolean,
       totals: Map<RecordedSpeciesKey, Map<RecordedPlantStatus, Int>>,
       cumulativeDeadFromCurrentObservation: Boolean = false,
+      survivalRateDensityCondition: Condition,
```

Comment:
I couldn't think of a better way to do this without having to pass it in to each. If someone knows of a way for this, I'm all ears.

Reply:
Can't think of a cleaner way offhand, given that you have to do different joins in each case.